crates/deps/gen/src/bstr.rs

```diff
@@ -48,40 +48,34 @@ pub fn gen_bstr() -> TokenStream {
                 Self::from_wide(self.as_wide())
             }
         }
-        // TODO: these str traits should work with no_std
-        #[cfg(feature = "std")]
         impl ::core::convert::From<&str> for BSTR {
             fn from(value: &str) -> Self {
-                let value: ::std::vec::Vec<u16> = value.encode_utf16().collect();
+                let value: ::windows::runtime::alloc::vec::Vec<u16> = value.encode_utf16().collect();
```

Comment:
@kennykerr Shouldn't this and other allocations for `BSTR` be wrapped in the `alloc` feature just like some of the other functions?

In other words, perhaps you should hide:
```rust
#[doc(hidden)]
pub extern crate alloc;
```
Behind the `alloc` feature too?

Reply:
The `alloc` feature is to avoid implicit allocations like `IntoParam`. `From`/`Into` should still work regardless as they are explicit.